xml/vk.xml

```diff
@@ -143,23 +143,23 @@ server.
         <type requires="ggp_c/vulkan_types.h" name="GgpFrameToken"/>
 
         <type category="define">#define <name>VK_MAKE_VERSION</name>(major, minor, patch) \
-    (((major) &lt;&lt; 22) | ((minor) &lt;&lt; 12) | (patch))</type>
-        <type category="define">#define <name>VK_VERSION_MAJOR</name>(version) ((uint32_t)(version) &gt;&gt; 22)</type>
-        <type category="define">#define <name>VK_VERSION_MINOR</name>(version) (((uint32_t)(version) &gt;&gt; 12) &amp; 0x3ff)</type>
+    (((major) &lt;&lt; 22U) | ((minor) &lt;&lt; 12U) | (patch))</type>
+        <type category="define">#define <name>VK_VERSION_MAJOR</name>(version) ((uint32_t)(version) &gt;&gt; 22U)</type>
+        <type category="define">#define <name>VK_VERSION_MINOR</name>(version) (((uint32_t)(version) &gt;&gt; 12U) &amp; 0x3ff)</type>
         <type category="define">#define <name>VK_VERSION_PATCH</name>(version) ((uint32_t)(version) &amp; 0xfff)</type>
 
         <type category="define">// DEPRECATED: This define has been removed. Specific version defines (e.g. VK_API_VERSION_1_0), or the VK_MAKE_VERSION macro, should be used instead.
-//#define <name>VK_API_VERSION</name> <type>VK_MAKE_VERSION</type>(1, 0, 0) // Patch version should always be set to 0</type>
+//#define <name>VK_API_VERSION</name> <type>VK_MAKE_VERSION</type>(1U, 0U, 0U) // Patch version should always be set to 0</type>
         <type category="define">// Vulkan 1.0 version number
-#define <name>VK_API_VERSION_1_0</name> <type>VK_MAKE_VERSION</type>(1, 0, 0)// Patch version should always be set to 0</type>
+#define <name>VK_API_VERSION_1_0</name> <type>VK_MAKE_VERSION</type>(1U, 0U, 0U)// Patch version should always be set to 0</type>
```

Comment:
Maybe just cast those to `uint32_t`? If it is 64 bit then gets downcasted to 32, some other overzealous tool might complain.

Reply:
Alright, updated it!

Reply:
FWIW, I don't think that should be necessary. IIRC, common 64-bit systems (I can't vouch for a Cray 2...) all have 32-bit int and 64-bit long long, with Microsoft using 32-bit longs and most other platforms using 64-bit longs. Unless we're somehow trying to run on a 16-bit system, 1U, 0U and 0U should be 32-bit, and I'd find it easier to read. I won't object, but my mild preference would be to avoid unnecessary (uint32_t) casting. Unless, I suppose, we've done it everywhere else.

https://en.cppreference.com/w/c/language/integer_constant

Reply:
@fluppeteer Well Vulkan does not use the old `int` system at all. Except in the macros.

`uint32_t` could be `int` or `long` per your own direction of thinking too. Some other overzealous tool might complain sooner or later. Technically, literal for `uint32_t` is `UINT32_C(42)`.

From C++ standpoint, it is important the result to have well defined unambiguous type (for function overloading, and for templating).

Reply:
Hmm. Fair enough, I suppose there might be a tool which complains about casting from long to int even when it knows it's not actually narrowing. (I might call that a _buggy_ tool...) Likewise, I'd hope any C++ wrapper would be happy with the obvious type promotions in this case rather than having overloaded it based on nuance of int. I don't feel strongly about it, though; it makes it a bit less legible, but a lot of the API can be accused of the same crime.